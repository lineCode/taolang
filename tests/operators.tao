function exponentiation() {
    let a;
    
    a = 3;
    a = a ** 2;
    println(a);

    a = 3;
    a **= 2;
    println(a);

    println();
}

function multiplication() {
    let a;
    let b;
    let c;

    a = 3;
    a = a * 3;
    println(a);
    a = 3;
    a *= 3;
    println(a);

    b = 8;
    b = b / 2;
    println(b); 
    b = 8;
    b /= 2;
    println(b); 

    c = 8;
    c = c % 3;
    println(c); 
    c = 8;
    c %= 3;
    println(c); 
}

function all() {
    exponentiation();
    multiplication();
}

function main() {
    all();
}
