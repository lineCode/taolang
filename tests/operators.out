9
9

9
9
4
4
2
2
